pkg/resources/konnectivity/sidecar.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package konnectivity
+
+import (
+	"fmt"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// ProxySidecar returns container that runs konnectivity proxy server as a sidecar in apiserver pods.
+func ProxySidecar(serverCount int32) (*corev1.Container, error) {
+	return &corev1.Container{
+		Name:            "konnectivity-server-container",
+		Image:           "us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-server:v0.0.24",
+		ImagePullPolicy: corev1.PullAlways,
+		Command: []string{
+			"/proxy-server",
+		},
+		Args: []string{
+			"--logtostderr=true",
+			"-v=100",
+			fmt.Sprintf("--cluster-key=/etc/kubernetes/pki/%s.key", resources.KonnectivityProxyTLSSecretName),
+			fmt.Sprintf("--cluster-cert=/etc/kubernetes/pki/%s.crt", resources.KonnectivityProxyTLSSecretName),
+			"--uds-name=/etc/kubernetes/konnectivity-server/konnectivity-server.socket",
+			fmt.Sprintf("--kubeconfig=/etc/kubernetes/kubeconfig/%s", resources.KonnectivityServerConf),
+			fmt.Sprintf("--server-count=%d", serverCount),
+			"--mode=grpc",
+			"--server-port=0",
+			"--agent-port=8132",
+			"--admin-port=8133",
+			"--health-port=8134",
+			"--agent-namespace=kube-system",
```

Comment:
shouldn't this be a user cluster namespace?

Reply:
https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/47a9234435b115cc764d76c8336057ea27ed0db6/cmd/server/app/options/options.go#L96

Interesting proxy-server accepts only one value there but we have agents deployed in seed-cluster in `cluster-somename` and one in `kube-system` on the user-cluster. 

Let me check again how this is being used.